### PR TITLE
Make bedevere work as a GitHub App

### PR DESCRIPTION
**Pull request by Mariatta**
Instead of using Oauth Token, use the GitHub App Installation Access Token. If the "installation" dict is passed in the webhook event, use it.

Made some changes to accommodate testing in personal CPython fork.

Checklist for deploying this app:
- [x] Choose a name for the GitHub App. (Can't use `bedevere-bot` since it's already taken). Suggestions: `bedevere-py`, `bedevere-app`
- [x] Create a GitHub App under the Python org ~~/ transfer mariatta's app to the Python org~~
- [x] Set environment variables in bedevere's Heroku app: `GH_APP_ID`, `GH_PRIVATE_KEY` (from the newly created GitHub App in the previous step)
- [ ] Deploy and test

### CODE REVIEW

**Comment by Mariatta** [comment]
I tested the app by deploying it to a personal heroku instance, and created a GitHub App on my account.
Sample how it works: https://github.com/Mariatta/cpython/pull/77

It shows the bot applying the "skip news" and "docs" labels to the PR.
Since it is a personal App, it does not have access to the CPython core devs list, and therefore it applies the "awaiting review" label. (It doesn't know that I'm a core developer).

**Comment by hugovk** [comment]
> * [ ]  Choose a name for the GitHub App. (Can't use `bedevere-bot` since it's already taken). Suggestions: `bedevere-py`, `bedevere-app`

`bedevere-app` sounds good: it's a GitHub app, so it's descriptive. `bedevere-py` could be almost anything.

File: bedevere/util.py [hunk 1]
```diff
@@ -227,10 +231,12 @@ def no_labels(event_data):
         return False
 
 
-async def get_pr_for_commit(gh, sha):
+async def get_pr_for_commit(gh, sha, repo_full_name=None):
     """Find the PR containing the specific commit hash."""
+    if not repo_full_name:
+        repo_full_name = "python/cpython"
```
Comment (picnic-sven) [inline bedevere/util.py:237]:
```suggestion
async def get_pr_for_commit(gh, sha, repo_full_name="python/cpython"):
    """Find the PR containing the specific commit hash."""
```

**Review by picnic-sven — COMMENTED** [review]
Super small review. Exciting feature ✨ !

**Comment by Mariatta** [comment]
@ewdurbin checking if you've had chance to review this. Thanks.

**Comment by ewdurbin** [comment]
Hi @Mariatta this looks ready to go. I think I'd prefer us to create a new GitHub application under the python org.

What [permission requests](https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/choosing-permissions-for-a-github-app) need to be configured for the application to work?

**Comment by Mariatta** [comment]
These are the settings I have on my test app:

General:
- Webhook: active
- Webhook URL: the url to bedevere's Heroku app
- Webhook Secret: the `GH_SECRET` env var in Heroku

Repository permissions:
- Commit statuses: Read and write
- Issues: Read and write
- Metadata: Readonly
- Pull requests: read and write
- the rest are no access (default)

Organization permissions:
- no access 

Account permissions:
- no access 

Subscribe to events
- Issues
- Pull request
- Repository
- Pull Request Review

Let me know if there's anything else needed, and also give me heads up when this is set up. I can help debug issues in case anything arises.

**Comment by ewdurbin** [comment]
OK @Mariatta everything is set. The app has been created. I'm going to see if we can make it work :)

**Comment by ewdurbin** [comment]
Looks like it's working! https://github.com/python/cpython/pull/109264 :)

**Pull request merged**